src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -962,6 +1010,32 @@ private void EndUpdating()
             }
         }
 
+        private void StartTextSearchTimer()
+        {
+            _textSearchTimer = new DispatcherTimer { Interval = TimeSpan.FromSeconds(1) };
+            _textSearchTimer.Tick += TextSearchTimer_Tick;
+            _textSearchTimer.Start();
+        }
+
+        private void StopTextSearchTimer()
+        {
+            if (_textSearchTimer == null)
+            {
+                return;
+            }
+
+            _textSearchTimer.Stop();
```

Comment:
I would invert the commands to prevent TextSearchTimer_Tick from being called while stopping the timer.

Reply:
👌